pkg/action/action.go

```diff
@@ -210,14 +222,144 @@ func (c *Configuration) renderResources(ch *chart.Chart, values chartutil.Values
+	for fileName, fileContent := range files {
+		for _, document := range reYamlDocumentSeparator.Split(fileContent, -1) {
+			if strings.TrimSpace(document) == "" {
+				continue
+			}
+			b.WriteString("---\n")
+			documentYaml, err := yamlSetTemporaryAnnotation(document, fileName)
+			if err != nil {
+				return nil, fmt.Errorf("cannot prepare a file for post-process: %w", err)
+			}
+			b.Write(documentYaml)
+		}
+	}
+
+	// Run post-processing.
+	b, err := pr.Run(b)
+	if err != nil {
+		return nil, errors.Wrap(err, "error while running post render on files")
+	}
+
+	postProcessedFiles := make(map[string]string)
+	// Split the post-processed stream into files.
+	for _, document := range reYamlDocumentSeparator.Split(b.String(), -1) {
+		if strings.TrimSpace(document) == "" {
+			continue
+		}
+		document, fileName, err := yamlCutTemporaryAnnotation(document)
 		if err != nil {
-			return hs, b, notes, errors.Wrap(err, "error while running post render on files")
+			return nil, fmt.Errorf("cannot reconstruct a file after post-process: %w", err)
+		}
+		existingDocument, ok := postProcessedFiles[fileName]
+		if ok {
+			postProcessedFiles[fileName] = existingDocument + "\n---\n" + string(document)
+		} else {
+			postProcessedFiles[fileName] = string(document)
 		}
 	}
 
-	return hs, b, notes, nil
+	return postProcessedFiles, nil
+}
+
+// yamlSetTemporaryAnnotation adds temporary annotation in yaml kubernetess object
+func yamlSetTemporaryAnnotation(document, annotationValue string) ([]byte, error) {
+	// Unmarshal the document.
+	var documentMap map[string]interface{}
+	err := yaml.Unmarshal([]byte(document), &documentMap)
+	if err != nil {
+		return nil, err
+	}
+
+	// Take the annotations map.
+	metadata, ok := documentMap["metadata"].(map[string]interface{})
+	if !ok {
+		// Document has no metadata.
+		metadata = make(map[string]interface{})
+	}
+
+	annotations, ok := metadata["annotations"].(map[string]interface{})
+
+	if !ok {
+		annotations = make(map[string]interface{})
+	}
+	annotations[tempAnnotationName] = annotationValue
+
+	metadata["annotations"] = annotations
+	documentMap["metadata"] = metadata
+
+	documentYaml, err := yaml.Marshal(documentMap)
+	if err != nil {
+		return nil, fmt.Errorf("cannot marshal a yaml document: %v", err)
+	}
+
+	return documentYaml, nil
+}
+
+// yamlCutTemporaryAnnotation cuts out an annotation text from yaml kubernetess object
+func yamlCutTemporaryAnnotation(document string) ([]byte, string, error) {
+	// Unmarshal the document.
+	var documentMap map[string]interface{}
+	err := yaml.Unmarshal([]byte(document), &documentMap)
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Take the annotations map.
+	metadata, ok := documentMap["metadata"].(map[string]interface{})
+	if !ok {
+		return nil, "", fmt.Errorf("document has no metadata: %s", document)
+	}
+
+	annotations, ok := metadata["annotations"].(map[string]interface{})
+
+	annotationValue := ""
```

Comment:
Nit: This variable would be more descriptive for future readers if it was called `fileName`